### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by rgbkrk**
Yay for tests!

While I was at it, made `get_renderer` not show its Python roots by changing it to `getRenderer`. :wink:

/cc @jdfreder

### CODE REVIEW

**Comment by rgbkrk** [comment]
Welp, our transformRichest is a bit strange since we can't go to fallback gracefully.

I'm off to get :ice_cream:, but I'll be back to this shortly!

**Comment by jdfreder** [comment]
Love the last commit, lol

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "lib/transformime.js",
   "scripts": {
     "prebuild": "mkdirp lib && mkdirp dist && npm run clean",
-    "test": "echo 'tests are cool, we should think about a testing framework'",
```
Comment (jdfreder) [inline package.json]: I see you deleted the "not cool" while I wasn't paying attention :angry: :anger:

Comment (rgbkrk) [inline package.json]: 😅

File: src/transformime.js [hunk 2]
```diff
@@ -92,7 +94,7 @@ export class Transformime {
      * @param  string mimetype the data type (e.g. text/plain, text/html, image/png)
      * @return {RendererBase} Matching renderer
      */
-    get_renderer(mimetype) {
```
Comment (jdfreder) [inline src/transformime.js]: Whoops my Python is showing, again... :relaxed:

File: test/transformimeTest.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
+import {assert, expect} from 'chai';
+
+var jsdom = require('jsdom');
```
Comment (jdfreder) [inline test/transformimeTest.js]: Why isn't this `import * as jsdom from 'jsdom'`?

File: test/transformimeTest.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        ];
+        this.t = new Transformime(this.renderers);
+        this.document = jsdom.jsdom();
+    });
+    describe('transform', function() {
+        it('should have called our DummyRender', function() {
+            var elPromise = this.t.transform("dummy-data", "transformime/dummy1", this.document);
+
+            elPromise.then((el) => {
+                assert.equal(this.dummyRenderer1.lastData, "dummy-data");
```
Comment (jdfreder) [inline test/transformimeTest.js]: Where is `lastData` defined and set?

Comment (rgbkrk) [inline test/transformimeTest.js]: I cheated with my mock and registered their last called argument.

Comment (rgbkrk) [inline test/transformimeTest.js]: oh ha. Those are gone now. Tests are passing, chaos is everywhere.

File: test/transformimeTest.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                    'application/zip': 'zippy'
+                };
+
+                this.t.fallbackRenderer = new DummyRenderer("fallback/test");
+                let elPromise = this.t.transformRichest(mimeBundle, this.document);
+
+                elPromise.then( () => {
+                    assert.equal(this.t.fallbackRenderer.lastData, 'omg this is going to fail');
+                    done();
+                });
```
Comment (jdfreder) [inline test/transformimeTest.js]: There's another way to do this too, it requires more characters, but less lines and is debatabley easier to read for someone with "asynchronous eyes on".

```js
elPromise.then( () => {
    assert.equal(this.t.fallbackRenderer.lastData, 'omg this is going to fail');
}).then(done);
```

**Comment by rgbkrk** [comment]
Woo. Now fully promisified.

**Pull request merged**